libvirt/resource_libvirt_domain.go

```diff
@@ -748,3 +759,36 @@ func newDiskForCloudInit(virConn *libvirt.VirConnection, volumeKey string) (defD
 
 	return disk, nil
 }
+
+func getDomainInterfaces(domain *libvirt.VirDomain) ([]libvirt.VirDomainInterface, error) {
+
+	// get all the interfaces using the qemu-agent, this includes also
+	// interfaces that are not attached to networks managed by libvirt
+	// (eg. bridges, macvtap,...)
+	interfaces := getDomainInterfacesViaQemuAgent(domain, true)
+	if len(interfaces) > 0 {
+		// the agent will always return all the interfaces, both the
+		// ones managed by libvirt and the ones attached to bridge interfaces
+		// or macvtap. Hence it has the highest priority
+		return interfaces, nil
+	}
```

Comment:
Isn't `qemu-agent` authoritative about the IP addresses? I mean that, if `qemu-agent` is installed and we do not get IPs, maybe we could just return here...